src/NUnitFramework/framework/Internal/Commands/TimeoutCommand.cs

```diff
@@ -94,19 +94,24 @@ public override TestResult Execute(TestExecutionContext context)
         {
             try
             {
-                var testExecution = RunTestOnSeparateThread(context);
-                if (Task.WaitAny(new Task[] { testExecution }, _timeout) != -1
-                    || _debugger.IsAttached)
+                var separateContext = new TestExecutionContext(context)
                 {
-                    context.CurrentResult = testExecution.GetAwaiter().GetResult();
-                }
-                else
-                {
-                    string message = $"Test exceeded Timeout value of {_timeout}ms";
+                    CurrentResult = context.CurrentTest.MakeTestResult()
+                };
+                var testExecution = Task.Run(() => innerCommand.Execute(separateContext));
+                var timedOut = Task.WaitAny(new Task[] { testExecution }, _timeout) == -1;
 
+                separateContext.CurrentResult.ApplyOutput(context.CurrentResult);
+
+                if (timedOut && !_debugger.IsAttached)
```

Comment:
Note to reviewer: I had to move the `Task.WaitAny` to it's own line in order to apply the test output afterwards. I ended up inverting this check here as part of that as I found it read clearer this way but I'm happy to instead change it back to something like:
```
if (!timedOut || _debugger.IsAttached)
```